server/apps/main/templatetags/custom_tags.py

```diff
@@ -32,3 +32,8 @@ def url_replace(request, field, value):
     query = request.GET.copy()
     query[field] = value
     return '?{}'.format(query.urlencode())
+
+@register.filter
+def public_toggle(items, index):
```

Comment:
I'm not 100% sure I understand what's going on here. What exactly does the public_toggle do/check for? :)

Reply:
This was used to condition on the 'Make public' fields. But it was hacky and I should have put more effort into coming up with a bettter alternative. I've made a more general filter for this now so all the toggle fields can be handled the same way on the form.

Reply:
Ah, so this is the bit which in this PR replaces the regular checkbox with the toggle switch, right? :)

Reply:
Yes; pretty much (the actual work is done in the template &mdash; which I know you know 😄 &mdash; but this helps simplify the conditioning).